packages/frontend-web/src/app/scenes/Settings/components/RuleRow/RuleRow.tsx

```diff
@@ -123,14 +134,39 @@ export class RuleRow extends React.Component<IRuleRowProps> {
     callback(e.target.value);
   }
 
+  @autobind
+  notifyWrapperOfActionChange(action: IModerationAction) {
+    const {
+      onModerateButtonClick,
+      rule,
+    } = this.props;
+    let raction: IRuleAction;
+    switch (action) {
+      case 'approve':
```

Comment:
Why not use the constants in this case statement?

Reply:
They are, in principal, different values - the strings are the client-side values (i.e. IModerationAction values), the constants are the server-side values.  In particular, the approve/ACCEPT difference is apparent here.  I could try to rationalise this, but it is a big bit of work.

Reply:
I've centralised the map between client values and server values.  It makes things much clearer.  At some future point I'll start removing the client values and just use the server values throughout.